### ISSUE TITLE
lld-macho should report conflict method from another category

### ISSUE THREAD

**Issue by oontvoo** [llvm/llvm-project#54912]
LD64 emits warnings such as:

`method 'foo' in category from path/to/pack.a cofnlicts with same method from another category`.

**Comment by int3** [comment]
Repro / more detail would be helpful :)

**Comment by oontvoo** [comment]
Sample repro:

```
// foo.mm
#import <Foundation/Foundation.h>

@interface MyClass : NSObject
@end

@implementation MyClass : NSObject
@end

@implementation MyClass (MyCategory)
- (void) myCategoryFunction {}
@end

@implementation MyClass (MyCategoryTwo)
- (void) myCategoryFunction {}
@end
```

In ObjC you have this concept of "category function", which allows you to provide multiple impls for a class. But if you have something like in the example, then the linker should give you a warning.
